### DISCUSSION TITLE
Opening the outgoing messages page takes ages

### DISCUSSION THREAD

**Discussion by dewisuryani** [postalserver/postal#3219] [Help with using Postal]
## Describe the bug

When I open the outgoing messages page, it takes a very long time, and sometimes the page doesn’t even open at all.

Path: [DASHBOARD-URL]/messages/outgoing
Current outgoing messages: 35,697,645

## To Reproduce

1. Go to 'DASHBOARD-URL'
2. Click on 'Messages'
3. Wait until the page opens

## Expected behaviour

To be able to open this page as quickly as possible, optimization is needed to retrieve data from the database.

**Comment by willpower232** [comment]
That is definitely an awful lot of messages, I don't think Postal was meant to handle that. Are they processing? Hopefully it isn't going to take 2 months to send them all :sweat_smile:
